### DISCUSSION TITLE
Problems with make after using stork.

### DISCUSSION THREAD

**Discussion by TLWise** [idaholab/moose#16988] [Q&A General] [answered]
After using stork to create an application I get the following error after using the command make -j4:

In member function 'virtual void PhysicsBasedPreconditioner::init()':
/home/twise/projects/moose/framework/src/preconditioners/PhysicsBasedPreconditioner.C:209:41: error: 'class libMesh::LinearImplicitSystem' has no member named 'get_system_matrix'; did you mean 'add_system_matrix'?
     preconditioner->set_matrix(u_system.get_system_matrix());
                                         ^~~~~~~~~~~~~~~~~
What should I do to fix this problem.

Thanks!

**Comment by GiudGiud** [comment]
Hello

Most libmesh-related compilation issues are usually resolved by updating libmesh. 
If you installed with ana/miniconda just run 'conda update --all' in the moose environment

Best,
Guillaume

**Reply by TLWise** [reply]
Thank you for the advice on this issue. I ran the 'conda update --all' and the software informed me that all the updates were current. I also ran 'conda update -n base -c defaults conda', which did some additional installs, but still results in the same errors with the 'make -j4' command.

**Reply by GiudGiud** [reply]
Did you have a previous moose install before this one? Any chance a LIBMESH_DIR environment variable is set to a previously installed libmesh?
Are you on the latest moose master commit?

**Reply by TLWise** [reply]
I did have a previous install that I believed was totally uninstalled, but may not have been properly done. What can I do to ensure there are not remnants of a previous install, outside of reinstalling Ubuntu?

**Reply by GiudGiud** [reply]
I would check MOOSE_DIR and LIBMESH_DIR (echo $MOOSE_DIR, echo ..) to see if they are not pointing somewhere from the previous installation.

Just to check, you ran 'conda update --all' after having activated the moose conda environment right?

**Reply by GiudGiud** [reply]
Also maybe check CPATH, C_INCLUDE_PATH, CPLUS_INCLUDE_PATH if you had set those (I dont think you would have) at the time

**Reply by TLWise** [reply]
Yes. I ran each command after having executed "conda activate moose". I will check each of the files you indicated to see if the are pointing to a previous install.

Thank you.

**Reply by TLWise** [reply]
echo $LIBMESH_DIR results in this:
/home/twise/miniconda3/envs/moose/libmesh

echo $MOOSE_DIR results in a null result. I believe this indicates the working directory is the directory.

**Reply by GiudGiud** [reply]
This all seems normal to me. I don't know what's wrong then. Someone else will have to weigh in.

**Comment by loganharbour** [comment, marked as answer]
You’ve updated libmesh but you haven’t updated moose. The libmesh you’ve updated to within conda is no longer with the moose you have. 

If you need help with this using git - let us know.

**Reply by TLWise** [reply]
Thank you. Where can I find the directions to remove an old installation of moose?

**Reply by loganharbour** [reply]
The following should work:
```
cd /home/twise/projects/moose
git pull
```

This should work if the `origin` remote of your moose installation is `idaholab/moose`, which will be the case if you used the getting started pages on mooseframework.org. `git pull` will fetch the new commits from idaholab, and merge them into the branch that you are currently on.

**Comment by TLWise** [comment]
Thank you for advising me on how to solve this issue. Prior to your last bit of advise, I had already executed the conda uninstall of moose. I then attempted to reinstall moose via conda create. That did not go so well. In the end I went back to the mooseframework web page and went down the list of instructions for installing moose. After the install  Make -j4 and run_test executed with no problems.


Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Reply by loganharbour** [reply]
Glad to help!

Here, it's important to note that our default conda instructions provide an environment for compiling moose (by providing its dependencies and a compiled version of libmesh). To update, you do not need to remove and "re-install" everything. You simply need to update conda with `conda update --all` and update your moose repository (typically with `git pull`).
